src/dsp_tools/commands/xmlupload/project_client.py

```diff
@@ -19,6 +19,10 @@ class ProjectInfo:
 class ProjectClient(Protocol):
     """Interface (protocol) for project-related requests to the DSP-API."""
 
+    con: Connection
+    shortcode: str
+    project_info: ProjectInfo | None
+
```

Comment:
I just added these for better typing support. The real implementation of the ProjectClient already had these fields before.